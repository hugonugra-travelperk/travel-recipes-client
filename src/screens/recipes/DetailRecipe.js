import {useNavigate, useLocation} from 'react-router-dom';
import {CardWrapper} from '../../components/base/Card'
import RecipeElement from '../../components/recipes/RecipeElement'
import {Button} from '../../components/base/Button'

export default function EditRecipe() {
  const location = useLocation();
  let navigate = useNavigate();

  let recipe = location.state.recipe;

  function goBack() {
    navigate('/')
  }

  return (
    <CardWrapper>
      <RecipeElement detail={true} recipe={recipe}/>
      <Button primary onClick={() => goBack()}> Back </Button>
    </CardWrapper>
  )
}
