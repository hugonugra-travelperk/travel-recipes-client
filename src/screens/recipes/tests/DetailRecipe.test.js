import {render, cleanup} from "@testing-library/react"
import "@testing-library/jest-dom";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import DetailRecipe from "../DetailRecipe"

const state = {
  recipe: {
    id: 1,
    name: "test text recipe",
    description: "test text description",
    ingredients: [{ id: 0, name: "test text ingredient" }],
  }
}

it("renders", async () => {
  const history = createMemoryHistory();
  history.push("/recipes/1", state);

  const {asFragment, findByText} = render(
    <Router location={history.location} navigator={history}>
      <DetailRecipe/>
    </Router>);
      
    expect(await findByText(state.recipe.name)).toBeVisible();
    expect(await findByText(state.recipe.description)).toBeVisible();
    expect(await findByText(state.recipe.ingredients[0].name)).toBeVisible();
});
