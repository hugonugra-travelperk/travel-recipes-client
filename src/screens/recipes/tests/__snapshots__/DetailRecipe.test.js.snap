// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL gaBRgU"
  >
    <div
      class="sc-bdvvtL gaBRgU"
    >
      <header
        class="sc-gsDKAQ jGoJmq"
      >
        test text recipe
      </header>
      <div
        class="sc-dkPtRN PHDyg"
      >
        test text description
        <header
          class="sc-hKwDye klnJXo"
        >
          Ingredients
        </header>
        <div
          class="sc-eCImPb gayBGf"
        >
          <ul>
            <li
              class="sc-jRQBWg kEHRpN"
            >
              <span
                class="sc-gKclnd CJqjG"
              >
                test text ingredient
              </span>
            </li>
          </ul>
        </div>
        <button
          class="sc-iCfMLu fQrYjb"
        >
          Edit
        </button>
        <button
          class="sc-iCfMLu gMEgxH"
        >
           Delete 
        </button>
      </div>
    </div>
    <button
      class="sc-iCfMLu gMEgxH"
    >
       Back 
    </button>
  </div>
</DocumentFragment>
`;
